client_test.go

```diff
@@ -0,0 +1,74 @@
+package client
+
+import (
+	"bytes"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHandleResponse(t *testing.T) {
```

Comment:
This only tests the new handleResponse method. Please let me know if I should also add tests for all the client methods (currently there are no existing tests so I would like some guidance before writing more tests).